cyclops/query/base.py

```diff
@@ -59,20 +78,34 @@ def __init__(self, table_map: Dict, column_map: Dict, **config_overrides) -> Non
 
         self._db = Database(config)
         self._table_map = table_map
-        self._column_map = column_map
+        self._setup_table_methods()
+
+    def list_tables(self) -> list:
```

Comment:
Would it be desireable to opt for a generator over a list here, to make it lazy?